integreat_cms/cms/forms/custom_content_model_form.py

```diff
@@ -230,7 +230,16 @@ def add_success_message(self, request: HttpRequest) -> None:
         :param request: The current request submitting the translation form
         """
         model_name = type(self.instance.foreign_object)._meta.verbose_name.title()
-        if not self.instance.status == status.PUBLIC:
+        if hasattr(self.instance, "page") and not request.user.has_perm(
+            "cms.publish_page_object", self.instance.page
+        ):
```

Comment:
This condition might break when we change the status rules in the form views. I'm not super happy about making assumptions about permissions here, since such an assumption was essentially the source of the bug. How about being a bit more clear here and making the message dependent on the new status?

```
        if self.instance.status == status.AUTO_SAVE:
            messages.success(
                request,
                _('{} "{}" was saved automatically').format(
                    model_name, self.instance.title
                ),
            )
        elif self.instance.status == status.REVIEW:
            messages.success(
                request,
                _('{} "{}" was successfully submitted for review').format(
                    model_name, self.instance.title
                ),
            )
        elif self.instance.status == status.DRAFT:
            messages.success(
                request,
                _('{} "{}" was successfully saved as draft').format(
                    model_name, self.instance.title
                ),
            )
        elif self.instance.status == status.PUBLIC:
            if "status" in self.changed_data:
                messages.success(
                    request,
                    _('{} "{}" was successfully published').format(
                        model_name, self.instance.title
                    ),
                )
            else:
                messages.success(
                    request,
                    _('{} "{}" was successfully updated').format(
                        model_name, self.instance.title
                    ),
                )
```

Reply:
Oh yeah, that is much better.